app/src/main/java/org/mozilla/fenix/compose/tabstray/TabListItem.kt

```diff
@@ -0,0 +1,185 @@
+
+        DescriptionSection(
+            modifier = Modifier.weight(descriptionWeight),
+            tab = tab
+        )
+
+        CloseIcon(
+            tab = tab,
+            visible = !multiSelectionEnabled,
+            modifier = Modifier.weight(1f),
+            onCloseClicked = onCloseClicked
+        )
+    }
+}
+
+@Composable
+private fun Thumbnail(
+    tab: TabSessionState,
+    multiSelectionEnabled: Boolean,
+    isSelected: Boolean,
+    modifier: Modifier,
+    onMediaIconClicked: ((TabSessionState) -> Unit)?
+) {
+    Box(
+        modifier = modifier
+            .padding(start = 16.dp, end = 12.dp, top = 8.dp, bottom = 8.dp)
+    ) {
+        TabThumbnail(
+            tab = tab,
+            modifier = Modifier.size(92.dp, 72.dp),
+            thumbnailDescription = stringResource(id = R.string.mozac_browser_tabstray_open_tab),
+            contentScale = ContentScale.FillWidth,
+            alignment = Alignment.TopCenter
+        )
+
+        if (isSelected) {
+            Card(
+                modifier = Modifier
+                    .size(40.dp)
+                    .align(Alignment.Center),
+                shape = CircleShape,
+                backgroundColor = colorResource(id = R.color.accent_normal_theme),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_check),
+                    modifier = Modifier
+                        .matchParentSize()
+                        .padding(8.dp),
+                    contentDescription = null,
+                    tint = colorResource(id = R.color.mozac_ui_icons_fill)
+                )
+            }
+        }
+        if (!multiSelectionEnabled) {
+            MediaImage(tab, onMediaIconClicked, Modifier.align(Alignment.Center))
+        }
+    }
+}
+
+@Composable
+private fun CloseIcon(
+    tab: TabSessionState,
+    visible: Boolean,
+    modifier: Modifier,
+    onCloseClicked: ((TabSessionState) -> Unit)?
+) {
+    if (visible) {
+        Column(
+            modifier = modifier
+                .fillMaxHeight()
+                .padding(
+                    top = 24.dp, bottom = 24.dp
+                ),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Image(
+                alignment = Alignment.Center,
+                colorFilter = ColorFilter.tint(colorResource(id = R.color.tab_tray_item_text_normal_theme)),
+                painter = painterResource(id = R.drawable.mozac_ic_close),
+                contentDescription = stringResource(
+                    id = R.string.close_tab_title,
+                    tab.content.title
+                ),
+                modifier = Modifier
+                    .width(48.dp)
+                    .clickable { onCloseClicked?.invoke(tab) }
+            )
+        }
+    }
+}
+
+@Composable
+private fun DescriptionSection(
+    modifier: Modifier,
+    tab: TabSessionState
```

Comment:
We might be able to simplify this by just passing in `title` and `url`. It is probably better not to pass in such big data class like TabSessionState to avoid recomposing when we only care about 2 string states.